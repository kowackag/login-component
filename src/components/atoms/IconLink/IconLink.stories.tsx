import type { Meta, StoryObj } from "@storybook/react";

import { IconLink } from "./IconLink";

const meta: Meta<typeof IconLink> = {
	component: IconLink,
};

export default meta;

type Story = StoryObj<typeof IconLink>;

export const Default: Story = {
	args: {
		href: "",
		name: "google",
	},
};
