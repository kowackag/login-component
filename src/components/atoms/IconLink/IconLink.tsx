export const IconLink = ({
	href,
	name,
}: {
	href: string;
	name: "facebook" | "linkedin" | "google";
}) => {
	const icons = {
		facebook: "/facebook.svg#facebook",
		google: "/google.svg#google",
		linkedin: "/linkedin.svg#linkedin"
	}
	return (
		<a className="mx-2 my-2" href={href} target="_black" aria-label={name}>
			<svg width="36px" height="36px">
				<use href={icons[name]}></use>
			</svg>
		</a>
	);
};
