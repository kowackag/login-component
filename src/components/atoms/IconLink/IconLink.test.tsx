import { expect, it, describe } from "vitest";
import { render, screen } from "@testing-library/react";
import { IconLink } from "./IconLink.tsx";

describe("IconLink", () => {
	it("link has appropriate attributes", async () => {
		render(<IconLink href="https://www.facebook.com/" name="facebook" />);

		const link = screen.getByLabelText("facebook");
		expect(link).toHaveAttribute("href", "https://www.facebook.com/");
		expect(link).toHaveAttribute("target", "_blank");
	});

	it("has appropriate icon", async () => {
		render(<IconLink href="https://www.linkedin.com/" name="linkedin" />);
		const svgEl = Array.from(screen.getByLabelText("linkedin").children)[0];
		const useEl = Array.from(svgEl.children)[0];
		expect(useEl).toHaveAttribute("href", "/linkedin.svg#linkedin");
	});
	it("renders correctly", () => {
		const wrapper = render(<IconLink href="https://www.facebook.com/" name="facebook" />);
		expect(wrapper.container).toMatchSnapshot();
	});
});
