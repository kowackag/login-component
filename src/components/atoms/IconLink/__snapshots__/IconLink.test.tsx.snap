// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconLink > renders correctly 1`] = `
<div>
  <a
    aria-label="facebook"
    class="mx-2 my-2"
    data-testid="link"
    href="https://www.facebook.com/"
    target="_blank"
  >
    <svg
      height="36px"
      width="36px"
    >
      <use
        href="/facebook.svg#facebook"
      />
    </svg>
  </a>
</div>
`;
